src/promptflow/promptflow/_sdk/entities/_trace.py

```diff
@@ -0,0 +1,91 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import json
+import typing
+
+from google.protobuf.json_format import MessageToJson
+from opentelemetry.proto.trace.v1.trace_pb2 import Span as PBSpan
+
+from promptflow._sdk._orm.trace import Span as ORMSpan
+from promptflow._sdk._utils import flatten_span_attributes
+
+
+class Span:
+    def __init__(
+        self,
+        span_id: str,
+        name: str,
+        span_type: str,
+        trace_id: str,
+        session_id: str,
+        content: str,
+        parent_span_id: typing.Optional[str] = None,
+        path: typing.Optional[str] = None,
+        run: typing.Optional[str] = None,
+        experiment: typing.Optional[str] = None,
+    ):
+        self.id = span_id
+        self.name = name
+        self.type = span_type
+        self.trace_id = trace_id
+        self.parent_span_id = parent_span_id
+        self.session_id = session_id
+        self.content = content
+        self.path = path
+        self.run = run
+        self.experiment = experiment
+
+    def _persist(self) -> None:
+        self._to_orm_object().persist()
+
+    @staticmethod
+    def _from_orm_object(obj: ORMSpan) -> "Span":
+        return Span(
+            span_id=obj.id,
+            name=obj.name,
+            span_type=obj.type,
+            trace_id=obj.trace_id,
+            parent_span_id=obj.parent_span_id,
+            session_id=obj.session_id,
+            content=json.loads(obj.content),
+            path=obj.path,
+            run=obj.run,
+            experiment=obj.experiment,
+        )
+
+    def _to_orm_object(self) -> ORMSpan:
+        return ORMSpan(
+            id=self.id,
+            name=self.name,
+            type=self.type,
+            trace_id=self.trace_id,
+            parent_span_id=self.parent_span_id,
+            session_id=self.session_id,
+            content=self.content,
+            path=self.path,
+            run=self.run,
+            experiment=self.experiment,
+        )
+
+    @staticmethod
+    def _from_protobuf_object(obj: PBSpan) -> "Span":
```

Comment:
I understand this function should also support instrumentation, so we might need some mechanism to keep collector sync with the exporter.